test/perf/perf_test.go

```diff
@@ -51,7 +60,38 @@ func TestPerformance(t *testing.T) {
 		err = host.WaitUntilMetricsCounterHasValue(metricsRoute.Status.Ingress[0].Host, "workqueue_depth", "name", "usersignup-controller", 0)
 		assert.NoError(t, err, "failed to reach the expected queue depth")
 		end := time.Now()
-		fmt.Printf("time to process the resource: %dms\n", end.Sub(start).Milliseconds())
+		logger.Info("done processing resources", "count", count, "duration (ms)", end.Sub(start).Milliseconds())
 	})
 
 }
+
+// initLogger initializes a logger which will write to `$(ARTIFACT_DIR)/perf-<YYYYMMDD-HHmmSS>.log` or `./tmp/perf-<YYYYMMDD-HHmmSS>.log` if no `ARTIFACT_DIR`
+// env var is defined.
+// Notes:
+// - the target directory will be created on-the-fly if needed
+// - it's up to the caller to close the returned file at the end of the tests
+func initLogger() (logr.Logger, *os.File, error) {
+	// log messages that need to be retained after the OpenShift CI job completion must be written in a file located in `${ARTIFACT_DIR}` to
+	var artifactDir string
+	if artifactDir = os.Getenv("ARTIFACT_DIR"); artifactDir == "" {
+		pwd, err := os.Getwd()
+		if err != nil {
+			return nil, nil, err
+		}
+		artifactDir = filepath.Join(pwd, "tmp")
```

Comment:
Should we use `TempDir()` instead of the current dir? Or maybe name the artifact dir something else? eg. 'perf-logs'?

Reply:
I usually have a `tmp` directory in my repo dir with uncommitted files, but related to the repo (logs I need to keep, etc.). But I can use `TempDir()` if it looks cleaner to you

Reply:
I guess `tmp` is okay since it's already git ignored and would indicate that it's fine to delete the directory.